src/rpc/evo.cpp

```diff
@@ -1616,16 +1616,16 @@ static UniValue protx(const JSONRPCRequest& request)
 #ifdef ENABLE_WALLET
     if (command == "protxregister" || command == "protxregister_fund" || command == "protxregister_prepare") {
```

Comment:
it seems as there's no way to warn user if he use deprecated RPC such as any message/etc, or does a way exist?

```
    if (command.size() > 5 && command.substr(command.size() - 5) == "_hpmn") {
        rpc_warning("Deprecated API! Use _evo suffix instead _hpmn");
    }

Reply:
Good idea! We can use `IsDeprecatedRPCEnabled` for this I guess. Also, I think `_hpmn` commands are actually broken now. So we need smth like
```diff
diff --git a/src/rpc/evo.cpp b/src/rpc/evo.cpp
index c8c8f9f37c..71897b5d86 100644
--- a/src/rpc/evo.cpp
+++ b/src/rpc/evo.cpp
@@ -793,6 +793,14 @@ static UniValue protx_register_evo(const JSONRPCRequest& request, const Chainsta
     bool isExternalRegister = request.strMethod == "protxregister_evo";
     bool isFundRegister = request.strMethod == "protxregister_fund_evo";
     bool isPrepareRegister = request.strMethod == "protxregister_prepare_evo";
+    if (request.strMethod.find("_hpmn") != std::string::npos) {
+        if (!IsDeprecatedRPCEnabled("hpmn")) {
+            throw JSONRPCError(RPC_METHOD_DEPRECATED, "*_hpmn methods are deprecated, set -deprecatedrpc=hpmn to enable them");
+        }
+        isExternalRegister = request.strMethod == "protxregister_hpmn";
+        isFundRegister = request.strMethod == "protxregister_fund_hpmn";
+        isPrepareRegister = request.strMethod == "protxregister_prepare_hpmn";
+    }
     return protx_register_common_wrapper(request, chainman, false, isExternalRegister, isFundRegister, isPrepareRegister, MnType::Evo);
 }
 
```